### PR TITLE
stm32: pinctrl: Add support for plain GPIO configuration

### LINKED ISSUE #42530: Possibility to define pinmux item for Pin Control as a plain input/output
**Is your enhancement proposal related to a problem? Please describe.**
My device driver uses pins which need to switch between different pin functions which seems to be perfectly suited job for Pin Control API, but I cannot find a way to switch pin between special function (i.e., UART TX ) and "simple" GPIO function (i.e., input with pull-up).
Unfortunately, it looks that Pin Control API allows **only** to switch between special pin functions which make it usable only in a very limited way for the cases like described above.

**Describe the solution you'd like**
`pinmux` nodes used to define `pinctrl-N` nodes should also accept configuration as normal GPIO pins i.e., `GPIO_INPUT | GPIO_PULL_UP`.

**Describe alternatives you've considered**
GPIO API does not offer configuration options for special pin functions, so this kind of function change can be only achieved using SOC specific low-level functions.

**Additional context**
My platform is STM32WB.

### PR DESCRIPTION
**Pull request by erwango**
Update STM32 pinctrl driver and dts bindings to allow configuration of pins as plain GPIO (Input, Output-low, Output-high).

Fixes #42530

### CODE REVIEW

**Comment by erwango** [comment]
@r2r0 ^^

File: dts/bindings/pinctrl/st,stm32-pinctrl.yaml [hunk 1]
```diff
@@ -78,6 +86,11 @@ child-binding:
             ... {
                      pinmux = <STM32_PINMUX('A', 9, ANALOG)>;
             };
+               GPIO A9 set as GPIO output high
+            ... {
+                     pinmux = <STM32_PINMUX('A', 9, ANALOG)>;
```
Comment (gmarull) [inline dts/bindings/pinctrl/st,stm32-pinctrl.yaml:91]: keeping `ANALOG` feels confusing here, shouldn't this be `GPIO`?

File: include/dt-bindings/pinctrl/stm32-pinctrl.h [hunk 2]
```diff
@@ -107,4 +109,10 @@
 #define STM32_PUPDR_MASK		0x3
 #define STM32_PUPDR_SHIFT		9
 
+/* GPIO plain output value */
+#define STM32_ODR_0			(0x0<<STM32_ODR_SHIFT)
+#define STM32_ODR_1			(0x1<<STM32_ODR_SHIFT)
```
Comment (gmarull) [inline include/dt-bindings/pinctrl/stm32-pinctrl.h:114]: nit: spaces around `<<`

File: include/dt-bindings/pinctrl/stm32f1-pinctrl.h [hunk 3]
```diff
@@ -108,4 +110,10 @@
 #define STM32_PUPD_MASK			0x3
 #define STM32_PUPD_SHIFT		7
 
+/* GPIO plain output value */
+#define STM32_ODR_0			(0x0<<STM32_ODR_SHIFT)
+#define STM32_ODR_1			(0x1<<STM32_ODR_SHIFT)
```
Comment (gmarull) [inline include/dt-bindings/pinctrl/stm32f1-pinctrl.h:115]: ditto

File: dts/bindings/pinctrl/st,stm32-pinctrl.yaml (hunk 1 above)
Comment (erwango) [reply dts/bindings/pinctrl/st,stm32-pinctrl.yaml:91]: oh, right... Will fix thanks

File: include/dt-bindings/pinctrl/stm32-pinctrl.h (hunk 2 above)
Comment (erwango) [reply include/dt-bindings/pinctrl/stm32-pinctrl.h:114]: And checkpatch.pl didn't catch this as it is busy tracking `_TypdeDef`..

**Review by gmarull — APPROVED** [review]
LGTM, I'll extend pinctrl guide about this option (static initialization of GPIO pins)

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**